### PR TITLE
comments, np.cumsum added to reply, disable extra prints

### PR DESCRIPTION
**Pull request by hobson**
* docstring style review

### CODE REVIEW

File: nlpia_bot/clibot.py [hunk 1]
```diff
@@ -169,8 +169,13 @@ def log_reply(self, statement, reply):
             json.dump(history, f)
 
     def reply(self, statement=''):
+        '''
+        MVP of class methods. Collects and selects from all possible replies from all loaded bots. Logs process. 
+        Returns chosen reply as string.  
+        '''
```
Comment (hobson) [inline nlpia_bot/clibot.py]: Could be a little more concise `""" Collect replies from from loaded bots and return best reply (str). """`

Comment (hobson) [reply nlpia_bot/clibot.py]: This is good. First and last sentence can be deleted, if you like. Also, I like to use `"` for natural language strings intended for humans and `'` for machine-readable strings.

File: nlpia_bot/clibot.py [hunk 2]
```diff
@@ -169,8 +169,13 @@ def log_reply(self, statement, reply):
...
         log.info(f'statement={statement}')
         replies = []
+        # Collect replies from each bot.
```
Comment (hobson) [inline nlpia_bot/clibot.py]: Looks good. I wouldn't spend too much time on comments, though, as they have less "leverage" than docstrings.

File: nlpia_bot/clibot.py [hunk 3]
```diff
@@ -187,19 +192,20 @@ def reply(self, statement=''):
                     log.error(str(e))
             bot_replies = normalize_replies(bot_replies)
             replies.extend(bot_replies)
+
+        # Weighted random selection of reply from those with top n confidence scores
```
Comment (hobson) [inline nlpia_bot/clibot.py]: This is a great comment. Code is complicated and comment simplifies it for the reader.

File: nlpia_bot/clibot.py [hunk 4]
```diff
@@ -187,19 +192,20 @@ def reply(self, statement=''):
...
             replies = sorted(replies, reverse=True)[:self.num_top_replies]
-            cumsum = 0
-            cdf = list()
-            for reply in replies:
-                cumsum += reply[0]
-                cdf.append(cumsum)
-            roll = np.random.rand() * cumsum
-            for i, threshold in enumerate(cdf):
+            
+            confidences, strings = list(zip(*replies))
```
Comment (hobson) [inline nlpia_bot/clibot.py]: I love it! Generally prefer `texts` over `strings` as a variable name for natural language text lists/arrays.

**Comment by KChalk** [comment]
Thanks guys. I fixed these in a more recent commit. Do I need to submit a new PR or amend this one somehow?

**Comment by hobson** [comment]
Awesome! It should automatically ammend the PR. I'll merge now.
--Hobson

**Pull request merged**